asm2vec/tensors.py

```diff
@@ -65,9 +88,20 @@ def calc_tensors(
                         torch.save(tensor, tensor_file)
                     else:
                         torch.save(tens.mean(0), tensor_file)
-                    tensors_list.append(entry.name)
 
     else:
         logging.info("No valid directory")
 
+    tensors_full_list = [f[:40] for f in os.listdir(tensor_path) if (not f.startswith('.'))]
+    tensors_list = list(set(tensors_full_list))
+
+    for tensor in tensors_list:
+        if tensors_full_list.count(tensor) > 1:
+            tensor_list = [torch.load(Path(os.path.join(tensor_path, x))).detach().squeeze() for x in
+                           os.listdir(tensor_path) if x[:40] == tensor]
+            torch.save(torch.mean(torch.stack(tensor_list), dim=0), os.path.join(tensor_path, tensor))
+            for fname in os.listdir(tensor_path):
+                if fname.startswith(f"{tensor}_"):
+                    os.remove(os.path.join(tensor_path, fname))
+
```

Comment:
1. Should this be a separate function? Such that you can easily add tests, etc?
2. Why `f[:40]`?